lib/seahorn/FiniteMapTransf.cc

```diff
@@ -0,0 +1,351 @@
+#include "seahorn/FiniteMapTransf.hh"
+
+#include "seahorn/Expr/ExprOpBind.hh"
+#include "seahorn/Expr/ExprCore.hh"
+#include "seahorn/Expr/ExprVisitor.hh"
+
+#include "seahorn/Support/SeaDebug.h"
+
+using namespace expr;
+using namespace expr::op;
+
+namespace {
+// just for checking if two sets of keys are the same, only use in debug mode
+template <typename T> bool compare(std::vector<T> &v1, std::vector<T> &v2) {
+  std::sort(v1.begin(), v1.end());
+  std::sort(v2.begin(), v2.end());
+  return v1 == v2;
+}
+
+  Expr mkVarGet(Expr map, Expr k, Expr vTy) {
+    // -- create a constant with the name get(map,k)
+    return bind::mkConst(variant::variant(0,finite_map::get(map, k)), vTy);
+  }
+
+  // \brief `m1` contains the same values as `m2`. Both maps are assumed to have
+  // the same keys `keys` but not necessarily in the same order, that is why
+  // `ks1` and `ks2` are needed.
+  Expr mkEqCore(Expr m1, Expr ks1, Expr m2, Expr ks2, ExprVector &keys, Expr vTy,
+                ExprFactory &efac, ExprSet &evars) {
+
+    ExprVector conj;
+
+    bool is_var_m1 = bind::isFiniteMapConst(m1);
+    bool is_var_m2 = bind::isFiniteMapConst(m2);
+
+    Expr e_m1, e_m2;
+
+    for (auto k : keys) {
+      if (is_var_m1) {
+        e_m1 = mkVarGet(m1, k, vTy);
+        evars.insert(e_m1);
+      } else
+        e_m1 = finite_map::mkGetVal(m1, ks1, k);
```

Comment:
I'm missing something. I though `m1` must be a finite map so that `mkGetVal` can succeed.